src/csharp/Microsoft.Spark.E2ETest/IpcTests/BroadcastTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.E2ETest.Utils;
+using Microsoft.Spark.Sql;
+using static Microsoft.Spark.Sql.Functions;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+    [Serializable]
+    public class TestBroadcastVariable
+    {
+        public int IntValue { get; private set; }
+        public string StringValue { get; private set; }
+
+        public TestBroadcastVariable(int intVal, string stringVal)
+        {
+            IntValue = intVal;
+            StringValue = stringVal;
+        }
+    }
+
+    [Collection("Spark E2E Tests")]
+    public class BroadcastTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public BroadcastTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _df = _spark.CreateDataFrame(new[] { "hello", "world" });
+        }
+
+        /// <summary>
+        /// Test Broadcast support by using multiple broadcast variables in a UDF.
+        /// </summary>
+        [Fact]
+        public void TestMultipleBroadcastWithoutEncryption()
+        {
+            var obj1 = new TestBroadcastVariable(1, "first");
+            var obj2 = new TestBroadcastVariable(2, "second");
+            Broadcast<TestBroadcastVariable> bc1 = _spark.SparkContext.Broadcast(obj1);
+            Broadcast<TestBroadcastVariable> bc2 = _spark.SparkContext.Broadcast(obj2);
+
+            Func<Column, Column> udf = Udf<string, string>(
+                str => $"{str} {bc1.Value().StringValue} and {bc2.Value().StringValue}");
+
+            var expected = new string[] {"hello first and second", "world first and second" };
+
+            DataFrame udfResult = _df.Select(udf(_df["_1"]));
+            string[] actual = DataFrameToString(udfResult);
+            Assert.Equal(expected, actual);
+        }
+
+        /// <summary>
+        /// Test Broadcast.Destroy() that destroys all data and metadata related to the broadcast
+        /// variable and makes it inaccessible from workers.
+        /// </summary>
+        [Fact]
+        public void TestDestroy()
+        {
+            var obj1 = new TestBroadcastVariable(5, "destroy");
+            Broadcast<TestBroadcastVariable> bc1 = _spark.SparkContext.Broadcast(obj1);
+
+            Func<Column, Column> udf = Udf<string, string>(
+                str => $"{str} {bc1.Value().StringValue}, {bc1.Value().IntValue}");
+
+            var expected = new string[] {"hello destroy, 5", "world destroy, 5"};
```

Comment:
Can you run ctrl+k ctrl+d to reformat the file? it should handles spaces for strings inside `{ }`